### PR TITLE
Motifs optional alphabet

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request removes alphabets as a requirement from Bio.motifs.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/motifs/__init__.py [hunk 1]
```diff
@@ -160,8 +160,12 @@ class Instances(list):
 
     def __init__(self, instances=None, alphabet=None):
         """Initialize the class."""
```
Comment (peterjc) [inline Bio/motifs/__init__.py]: The docstring ought to explain the arguments, especially as you are changing how the alphabet argument is interpreted.

File: Bio/motifs/__init__.py [hunk 2]
```diff
@@ -160,8 +160,12 @@ class Instances(list):
...
-        from Bio.Alphabet import IUPAC
         from Bio.Seq import Seq
+        try:
+            # Received an old-style alphabet
+            alphabet = alphabet.letters
```
Comment (peterjc) [inline Bio/motifs/__init__.py]: Could you call the internal variable letters instead of alphabet here? I would find that clearer.

**Comment by peterjc** [comment]
This seems premature without a wider decision on removing or simplifying alphabets.

I don't object to allowing passing a string of letters as an alternative input to an ``Alphabet`` object, but when you return a ``Seq`` object, could you not continue to set an alphabet object (even a generic one based on the letters in use)?

**Comment by mdehoon** [comment]
@peterjc  The decision to remove alphabets has already been made.

**Comment by peterjc** [comment]
@mdehoon has it? Have I missed some important emails?

**Comment by mdehoon** [comment]
@peterjc  Now that Bio.Alphabet has been declared obsolete, can we proceed with this pull request?

**Comment by peterjc** [comment]
If this just "removes alphabets as a requirement from Bio.motifs" I'd be happier, but it seems to go further and "removes Bio.Alphabets usage from Bio.motifs".

In particular, if I am reading the diff correctly, the ``Instances`` object's ``.alphabet`` property changes from a ``Bio.Alphabets`` object to a simple string. If you wanted to reduce the risk of breaking people's code, you could keep ``.alphabet`` as is for now (with a deprecation warning perhaps), and introduce ``.letters`` as a new string attribute?

Or if you are happy with a backward incompatible change, I would be OK with removing the old object attribute ``.alphabet`` and replacing it with a new string attribute ``.letters`` instead?

To me that naming is clearer, since ``.alphabet`` could just as easily mean the type (DNA/RNA/Protein/...) rather than the characters or letters expected, whereas we have the ``.letters`` convention well established.

**Comment by peterjc** [comment]
Also, could we merge #1453 first, and then update this pull request to apply the needed alphabet changes in the new motif parsers that adds?

**Comment by mdehoon** [comment]
Starting a new pull request

**Pull request closed without merging**